bin/chef-cluster

```diff
@@ -0,0 +1,127 @@
+#!/bin/sh
+#
+# Copyright 2017, SUSE
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+cmd=$1
+workercount=${2:-auto}
+vhostconfig=/etc/apache2/vhosts.d/chef-cluster.conf
+serverconfig=/etc/sysconfig/chef-server
+webroot=/srv/www/chef-cluster
+
+baseport=4001
+
+balancerstart=BEGIN-BALANCER-MEMBER
+balancerend=END-BALANCER-MEMBER
+
+usage() {
+  echo "usage: chef-cluster <enable|disable> [<workercount>|auto]"
+  exit 1
+}
+
+case $cmd in
+  enable)
+    [ "$workercount" == "auto" ] && workercount=$(grep -c ^processor /proc/cpuinfo)
+    [ "$workercount" -ge 2 ] || { echo "ERROR: workercount must be >= 2"; exit 2; }
+
+    # apache configuration
+    mkdir -p "$webroot"
+    a2enmod -q proxy_balancer || a2enmod proxy_balancer
+    a2enmod -q lbmethod_byrequests || a2enmod lbmethod_byrequests
+    tmpvhostconfig=$(mktemp)
+    test -e "$vhostconfig" && cat "$vhostconfig" > "$tmpvhostconfig" || cat <<EOF > $tmpvhostconfig
```

Comment:
`cat "$vhostconfig" > "$tmpvhostconfig"` is interesting, I would do `cp "$vhostconfig" "$tmpvhostconfig"` ;-)

I understand why you want to keep that; I kind of disagree, but I won't push too hard. My only request is that in the case of re-using the existing config file, we add some `echo` to explain we're reusing it.

Hmm, but actually: what happens if the user runs `chef-cluster enable 2` and then `chef-cluster enable 4`?

Reply:
`chef-cluster enable 4` will re-configure the whole thing to use 4 workers. That's the whole point for re-using existing vhost file.
Use case:
1. `chef-cluster enable 2` (or `auto` or whatever)
2. test the whole thing, modify vhost file a bit (e.g. expose `balancer-manager` or change some timeouts)
3. decide to add more workers
4. `chef-cluster enable 4` will do exactly that without losing any changed done to vhost file.
Same applies when you want to remove some workers.